google/generativeai/types/retriever_types.py

```diff
@@ -144,24 +144,25 @@ def to_state(x: StateOptions) -> State:
 
 @string_utils.prettyprint
 @dataclasses.dataclass
-class MetadataFilters:
+class MetadataFilter:
     key: str
-    conditions: Condition
+    conditions: Iterable[Condition]
 
 
 @string_utils.prettyprint
 @dataclasses.dataclass
 class Condition:
     value: str | float
+    operation: Operator
 
 
 @string_utils.prettyprint
 @dataclasses.dataclass
 class CustomMetadata:
```

Comment:
Give this class a ._to_proto method.

Reply:
Maybe this class is clearer if it's just `key` + `value` 

Then _to_proto can:

kwargs ={}
if isinstance(self.value, str):
  kwargs['string_value'] = self.value
elif isinstance(self.value, iterable):
  kwargs['string_list_value'] = glm.StringList(values=self.value)
elif isinstance(self.value, (int, float)):
  kwargs['numeric_value'] = float(self.value)
else:
  error!

return glm.CustomMetadata(key=self.key, **kwargs) ```